### ISSUE TITLE
Workflow shutdown not triggering actual shutdown

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-server#766]
I'm testing the new "Cancel" function on workflows. With the 1.3.1 update, I don't get an error anymore in Fractal web and I see the following in the server log .out:

```
172.27.29.116:52334 - "GET /api/v1/project/7/job/9/stop/ HTTP/1.1" 200
172.27.29.116:52326 - "GET /auth/whoami HTTP/1.1" 200
172.27.29.116:52334 - "GET /api/alive/ HTTP/1.1" 200
172.27.29.116:52326 - "GET /auth/whoami HTTP/1.1" 200
172.27.29.116:52348 - "GET /api/v1/project/7 HTTP/1.1" 200
172.27.29.116:52326 - "GET /api/v1/project/7/workflow/ HTTP/1.1" 200
172.27.29.116:52334 - "GET /api/v1/project/7/job/ HTTP/1.1" 200
```

So the stop gets sent to the server, nothing shows up in the .err log. But the actual shutdown doesn't appear to be working, jobs keep running for another minute, then the next job starts.
I'll need to test with longer running workflows to see if it would eventually get shut down and the jobs just happen to finish within a minute for those tasks, thus never run into it.
A minute is quite long though (do we know how often the executor checks for this?). I'll report updates here when I have a slower running example to test

**Comment by tcompa** [comment]
Can you check whether in the server-side job folder (the ones with files like `0.args.json`, ...) there is a file named `shutdown`?

**Comment by tcompa** [comment]
> do we know how often the executor checks for this?

Much more frequently than a minute. It's closer to one second.

**Comment by jluethi** [comment]
Yes, there is a file called shutdown, with the content:
```
Trigger executor shutdown for job.id=9, project_id=7.
```

The IDs are correct for the project & job. But nothing happened on slurm.

Let's also test at UZH once that deployment is ready. Does the shutdown get triggered by the actual user? Do we see the command somewhere it would use?

**Comment by tcompa** [comment]
> Yes, there is a file called shutdown, with the content: [..]

OK, thanks. This is as expected.

> Does the shutdown get triggered by the actual user?

No, by the `fractal` user running the server. Which should have the right `sudo` permissions for calling `scancel` as a user (does it?).

> Do we see the command somewhere it would use?

The code that builds the command is here https://github.com/fractal-analytics-platform/fractal-server/blob/0c26c0c343342b16efe2446de6353b9b5e8f5beb/fractal_server/app/runner/_slurm/executor.py#L1103-L1105

This constructs the correct command, in the tests, but something may be wrong in the current situation.

I would expect multiple logs appearing in fractal-server, since they are at INFO of WARNING level, but maybe that's a logging issue (ref https://github.com/fractal-analytics-platform/fractal-server/issues/620).

**Comment by jluethi** [comment]
> No, by the fractal user running the server. Which should have the right sudo permissions for calling scancel as a user (does it?).

Ah, that may be the issue then! On the FMI side, the fractal user can't even submit to the cluster. Also on the UZH side, not sure that would work this way. We may need to impersonate the user to cancel such jobs.

Also tested it in the UZH deployment and there, jobs also don't get shut down.

**Comment by jluethi** [comment]
Ah, I misunderstood then.

It's also not working at UZH for me though. Let's debug it there when we get a chance, given that the permission for scancel should already be there

**Comment by tcompa** [comment]
I found the bug (the `shutdown_file` attribute was not set to a reasonable default), and it will be fixed by #768 (waiting for the CI).

I released `1.3.2a0` for testing, ran example 01 from fractal-demos, cancelled the job from fractal-web, and finally observed the expected behavior (through fractal-web, but this is not relevant):
```
Workflow Job logs

JOB ERROR:
TRACEBACK:
JobExecutionError

Job cancelled due to executor shutdown.
```

As soon as #768 is merged, I'll release 1.3.2 with the bugfix.

**Comment by tcompa** [comment]
Re-opening as CI randomly failed in `main`.
